File: server/models/Vigile.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');

const vigileSchema = new mongoose.Schema({
  nom: {
    type: String,
    required: true
  },
  prenom: {
    type: String,
    required: true
  },
  telephone: {
    type: String,
    required: true,
    unique: true
  },
  code_acces: {
    type: String,
    required: true
  },
  poste: {
    type: String,
    default: 'porte_principale'
  },
  statut: {
    type: String,
    enum: ['actif', 'inactif'],
    default: 'actif'
  }
}, {
  timestamps: true
});

vigileSchema.pre('save', async function(next) {
  if (!this.isModified('code_acces')) return next();
  this.code_acces = await bcrypt.hash(this.code_acces, 10);
  next();
});

module.exports = mongoose.model('Vigile', vigileSchema); 

File: eduflex_mobile/lib/models/vigile.dart
class Vigile {
  final String id;
  final String nom;
  final String prenom;
  final String telephone;
  final String poste;
  final String statut;

  Vigile({
    required this.id,
    required this.nom,
    required this.prenom,
    required this.telephone,
    required this.poste,
    required this.statut,
  });

  factory Vigile.fromJson(Map<String, dynamic> json) {
    return Vigile(
      id: json['_id'] ?? json['id'] ?? '',
      nom: json['nom'] ?? '',
      prenom: json['prenom'] ?? '',
      telephone: json['telephone'] ?? '',
      poste: json['poste'] ?? 'porte_principale',
      statut: json['statut'] ?? 'actif',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'prenom': prenom,
      'telephone': telephone,
      'poste': poste,
      'statut': statut,
    };
  }
}
